kvbc/src/sparse_merkle/proof_path_processor.cpp

```diff
@@ -20,37 +20,39 @@ namespace sparse_merkle {
 namespace proof_path_processor {
 
 using Sliver = concordUtils::Sliver;
+using boost::container::static_vector;
 
-bool verifyProofPath(Sliver key, Sliver value, const std::vector<Hash>& proofPath, const Hash& rootHash) {
+bool verifyProofPath(Sliver key,
+                     Sliver value,
+                     const static_vector<Hash, Hash::SIZE_IN_BITS>& proofPath,
+                     const Hash& rootHash) {
   Hasher hasher{};
   auto keyHash = hasher.hash(key.data(), key.length());
   auto valueHash = hasher.hash(value.data(), value.length());
 
   enum class Direction : std::uint8_t { Left, Right };
   constexpr auto BatchedInternalNodeLevelsIterated = BatchedInternalNode::MAX_HEIGHT - 1;
 
-  std::vector<Direction> pathOrdering;
-  std::array<Direction, BatchedInternalNodeLevelsIterated> subPath;
+  static_vector<Direction, Hash::SIZE_IN_BITS> pathOrdering;
+  static_vector<Direction, BatchedInternalNodeLevelsIterated> subPath;
   for (size_t nibble_index = 0; nibble_index < Hash::MAX_NIBBLES; nibble_index++) {
     auto current_nibble = keyHash.getNibble(nibble_index).data();
     auto node_index = BatchedInternalNode::nibbleToIndex(current_nibble);
-    size_t foundElemens = 0;
+    subPath.clear();
```

Comment:
Why is this needed?

Reply:
This clearing is needed, because we reuse the same container when we iterate to extract the directions based on the nibbles of the key for each sub-record of the tree.